modules/contracts/src.ts/commitments/withdraw.ts

```diff
@@ -113,16 +113,22 @@ export class WithdrawCommitment {
         continue;
       }
       let recovered: string;
-      try {
-        recovered = await recoverAddressFromChannelMessage(hash, sig);
-      } catch (e) {
-        recovered = e.message;
-      }
-      if (recovered !== this.alice && recovered !== this.bob) {
-        throw new Error(`Invalid signer detected. Got ${recovered}, expected one of: ${this.alice} / ${this.bob}`);
+      if (sig === mkSig("0x0")) {
+        this.aliceSignature = this.alice ? sig : this.aliceSignature;
+        this.bobSignature = this.bob ? sig : this.bobSignature;
+      } else {
+        try {
+          recovered = await recoverAddressFromChannelMessage(hash, sig);
+        } catch (e) {
+          recovered = e.message;
+        }
+        if (recovered !== this.alice && recovered !== this.bob) {
+          throw new Error(`Invalid signer detected. Got ${recovered}, expected one of: ${this.alice} / ${this.bob}`);
+        }
+
+        this.aliceSignature = recovered === this.alice ? sig : this.aliceSignature;
+        this.bobSignature = recovered === this.bob ? sig : this.bobSignature;
```

Comment:
I don't like this logic being with the commitment, the commitment (in my view) is something that should always be redeemable onchain. Are we *sure* this needs to be here? Shouldn't cancellations be handled outside of the commitment class